jscomp/core/js_packages_info.ml

```diff
@@ -183,32 +199,35 @@ let get_output_dir (info : t) ~package_dir module_system =
   Filename.concat package_dir (get_js_path info module_system)
 
 let add_npm_package_path (packages_info : t) (s : string) : t =
-  if is_empty packages_info then
-    raise (Arg.Bad "please set package name first using -bs-package-name")
-  else
-    let handle_module_system module_system =
-      match module_system_of_string module_system with
-      | Some x -> x
-      | None -> raise (Arg.Bad ("invalid module system " ^ module_system))
-    in
-    let m =
-      match Ext_string.split ~keep_empty:true s ':' with
-      | [ path ] -> { module_system = NodeJS; path; suffix = Js }
-      | [ module_system; path ] ->
-          {
-            module_system = handle_module_system module_system;
-            path;
-            suffix = Js;
-          }
-      | [ module_system; path; suffix ] ->
-          {
-            module_system = handle_module_system module_system;
-            path;
-            suffix = Ext_js_suffix.of_string suffix;
-          }
-      | _ -> raise (Arg.Bad ("invalid npm package path: " ^ s))
-    in
-    { packages_info with module_systems = m :: packages_info.module_systems }
+  let handle_module_system module_system =
+    match module_system_of_string module_system with
+    | Some x -> x
+    | None -> raise (Arg.Bad ("invalid module system " ^ module_system))
+  in
+  let m =
+    match Ext_string.split ~keep_empty:true s ':' with
+    | [ path ] -> { module_system = NodeJS; path; suffix = Js }
+    | [ module_system; path ] ->
+        {
+          module_system = handle_module_system module_system;
+          path;
+          suffix = Js;
+        }
+    | [ module_system; path; suffix ] ->
+        {
+          module_system = handle_module_system module_system;
+          path;
+          suffix = Ext_js_suffix.of_string suffix;
+        }
+    | _ -> raise (Arg.Bad ("invalid npm package path: " ^ s))
+  in
+  { packages_info with module_systems = m :: packages_info.module_systems }
+
+let bs_module_type_sentinel = "MELANGE_INVALID_PACKAGE_OUTPUT"
+
+let add_npm_module_system (packages_info : t) module_system =
+  let m = { module_system; path = bs_module_type_sentinel; suffix = Js } in
```

Comment:
I wonder if there could be any issues later on due to the path being set to this sentinel value, I guess we rely here on the constraint where Dune private libs can't be consumed from public ones, right? Or what's the `path` used for?

Reply:
This is a bit unfortunate, but it's here because I want to support 2 workflows, and I don't wanna break the CLI interface just yet.

1. the flow where you emit (possibly multiple) JS files next to the `.cmj`
2. the flow where those (possibly multiple) package specs are generated one JS file at a time (implies CMJ generation with `-bs-stop-after-cmj`

Reply:
I think that, in the future, we can separate all these concerns into a better command line interface:

1. "batch" mode gets (potentially multiple) `-bs-package-output commonjs:./folder:.bs.js` 
2. "separate compilation" mode gets `-bs-path-inside-package folder -bs-stop-after-cmj` and `-bs-module-type es6 -o foo.mjs`